### DISCUSSION TITLE
Testing the Common-Backend-Scripts natively in USS and through a Jenkins agent

### DISCUSSION THREAD

**Discussion by Jack-Billings-IBM** [IBM/dbb#201] [Q&A]
Hello,

I'm trying to set up a pipeline through Jenkins that will leverage the Common-Backend-Scripts. To get started, I cloned this DBB repository into a USS directory on my test system (Wazi aaS instance) and configured the pipelineBackend.config file. I then tried running the gitClone.sh command natively in USS with this command: /usr/lpp/IBM/dbb/pipelineBackend/gitClone.sh -w /u/ibmuser/jenkins-remote/workspace/zos-test -r https://github.com/Jack-Billings-IBM/jack-dbb-zappbuild.git -b zAppBuild_2_x

It did clone my repository but I see some errors in the output. Errors include a syntax error in pipelineBackend.config and the WorkDir variable not being set. Exact pipelineBackend.config error:
/usr/lpp/IBM/dbb/pipelineBackend/pipelineBackend.config: line 1: unexpected EOF while looking for matching ``'
/usr/lpp/IBM/dbb/pipelineBackend/pipelineBackend.config: line 2: syntax error: unexpected end of file

These errors seem to be affecting the dbbBuild.sh script as it can't find the working directory.

A couple of questions:
1. Is it okay to run these Common-Backend-Scripts natively in USS? Is there a better way to test them?
2. Any guidance on fixing the pipelineBackend.config error?
3. What is the recommendation on running these scripts using a Jenkins agent? Any examples?

Note: I'm using DBB v1.1.3

**Comment by dennis-behm** [comment]
Hi @Jack-Billings-IBM ,

Yes, these Backend-scripts can also be used in a Jenkins pipeline context to simplify the invocation of the various pipeline stages. However, Jenkins has the ability to run an executor/node on USS, and you can benefit of the existing Jenkins plugin ecosystem.

Typically you would implement this encapsulation (that what the backend script do) within a [Jenkins shared library](https://www.jenkins.io/doc/book/pipeline/shared-libraries/). However, we have customers that use Jenkins but use an executor/node on a x86 box, where the backend script become relevant to manage and control the various stages/tasks of the pipeline.

To your issue. This looks like a encoding problem
> /usr/lpp/IBM/dbb/pipelineBackend/pipelineBackend.config: line 1: unexpected EOF while looking for matching ``'
/usr/lpp/IBM/dbb/pipelineBackend/pipelineBackend.config: line 2: syntax error: unexpected end of file

Can you please double check using `ls -lTr`, that the config file is correctly tagged? It should be `IBM-1047`
>t IBM-1047    T=on  -rw-r--r--   1 BPXROOT  TIVUSR     12769 Dec 18 13:04 pipelineBackend.config

The script have been tested with DBB 2.x; please keep in mind, that DBB 1.x will run out of support soon.
